bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/AbstractExpression.java

```diff
@@ -240,11 +272,36 @@ public Date getFinalFireTime() {
             }
         }
 
-        if (getCandidates().isEmpty()) {
-            return null;
+        Date lastCandidate = null;
+
+        if (!getCandidates().isEmpty()) {
+            if (getCandidates().size() == 1) {
+                lastCandidate = getCandidates().get(0);
+            } else {
+
+                while (getCandidates().size() == maximumCandidates) {
```

Comment:
I try to understand why you try to differentiate between floating start date and non-floating start date wrt getfinalfiretime()....